### ISSUE TITLE
[floating-ui react] Is there an example to create a tooltip with floating-ui?

### ISSUE THREAD

**Issue by StErMi** [floating-ui/floating-ui#1498]
I saw on the homepage of the project that you have an example that showcases that is possible to create tooltips with the `floating-ui` lib. 

I've dug the source code but as far as I see you are using `Tippy` internally to create that example.

So the question is if I need a Tooltip component for my app should I use Tippy or is there a way to recreate it with floating-ui and go with that?

**Comment by atomiks** [comment]
If you just want to add some tooltips and popovers right now and can't spend time crafting a custom solution from scratch, then Tippy is what you'll want for now. We're working on creating something like Tippy for Floating UI though.

**Comment by StErMi** [comment]
> If you just want to add some tooltips and popovers right now and can't spend time crafting a custom solution from scratch, then Tippy is what you'll want for now. We're working on creating something like Tippy for Floating UI though.
> 
> But the tutorial does show you how to build a tooltip using Floating UI [floating-ui.com/docs/tutorial](https://floating-ui.com/docs/tutorial)

yeah after re-reading tippy I saw that it's still using popper-v3 so I thought that it was only a matter of time ;)

**Comment by Slapbox** [comment]
> We're working on creating something like Tippy for Floating UI though.

How's the progress on this going? I need portaling for the tooltips with Tippy doesn't support. I looked over the docs, but creating the simplest functional tooltip is a daunting task it seems.

**Comment by atomiks** [comment]
It's come a long way since initial release in March last year, especially after adding the [`useTransition` hook](https://floating-ui.com/docs/useTransition) which brings it to being closer to Tippy's ease of use. 

Arrows are the main annoying thing rn that need a good API (there's an open issue). Aside from that, I don't think it's too hard to follow the [Tooltip guide](https://floating-ui.com/docs/tooltip). The reusable component example can be copy/pasted and worked on top of pretty easily.

**Comment by Slapbox** [comment]
@atomiks since you were the creator of Tippy; surely the FloatingUI Tooltip guide strikes you as more complex than Tippy? And that's what we're aiming to replace.

Tippy was nearly a drop-in solution. To make something like Tippy using FloatingUI takes hours, and there's various features like `outsidePress` to close the tooltip which we can't get to work. After most of a day's work trying to transition, I still feel I have at least another full day remaining.

I guess I'd just expect making tooltips to be easier with FloatingUI after all these years of Tippy's README suggesting it for new projects at this point.

**Comment by atomiks** [comment]
Yeah it's more complex but intentionally. It's meant to be super flexible to create all types of floating UI components in an accessible manner (which Tippy completely fails at for anything other than tooltips). Not to mention Tippy just isn't as flexible and works outside of React's paradigm which causes various problems.

If you have API feedback, feel free to post it in this issue: https://github.com/floating-ui/floating-ui/issues/2100. There's already been some feedback about the configuration complexity which I'll try to sort out before the final v1 release to make it simpler without sacrificing its flexibility (hopefully).

> There's various features like `outsidePress` to close the tooltip which we can't get to work. 

Can you explain what problems you're having? It should just work by default by placing `useDismiss(context)` into the `useInteractions` hook, without any more work required.

**Comment by Slapbox** [comment]
I mean, maybe I don't get how `outsidePress` is supposed to work. I'd expect it to trigger `onOpenChange` - but it doesn't seem to.

```js
  const context = useFloating({
    placement: 'bottom',
    middleware: [offset(10)],
    open: isOpen,
    onOpenChange: bool => {
      console.log(bool);
      return setIsOpen();
    },
  });

  const dismiss = useDismiss(context, {
     outsidePress: true,
  });

  const { getReferenceProps, getFloatingProps } = useInteractions([dismiss]);
```

Other issues (which I'll crosspost to the feedback issue you linked). From the docs:

- I'm not clear on how to configure the arrow with React
- I'm not clear on how to configure autoupdate with React
- I'm not clear on how to transition tooltips into view - for example in a situation where a single tooltip is shared across elements and it's already open on ElementA, but when ElementB is clicked we want it to fade in, not appear (achievable with keyframes but what a pain, is there a better way?)

Don't get me wrong at all, FloatingUI is amazingly powerful and like what I'd have dreamed of several years ago - but for building more powerful popper elements, not simple tooltips. The page for Tippy is redirecting users to FloatingUI, but it's still incredibly difficult to build a comparable tooltip in.

It's been years since that notice went up but FloatingUI really does not provide a good way to implement tooltips still. After all these years of that notice, I'd expect FloatingUI to be roughly as easy to use as Tippy, but it's actually one of the most difficult libraries to use in my experience because there is nothing available preconfigured - you have to build it all from a half dozen hooks that you need to (at least sort of) understand each of.

**Comment by atomiks** [comment]
I notice a couple of issues: `useFloating` returns an object that _has_ a `context` property, which is what the hooks accept. Not the return value of `useFloating` directly. The `bool` also needs to be passed to `setIsOpen`.

```diff
-  const context = useFloating({
+  const {context} = useFloating({
    placement: 'bottom',
    middleware: [offset(10)],
    open: isOpen,
    onOpenChange: bool => {
      console.log(bool); // Nothing is logged?
-      return setIsOpen();
+      return setIsOpen(bool);
    },
  });

  // `outsidePress: true` is already the default
  const dismiss = useDismiss(context);
```

> I'm not clear on how to configure the arrow with React

Yeah arrows in general are a pain. I need to provide an example but also make creating arrows easier. I'll need to add a temp example to the CodeSandboxes before we create an `<Arrow />` component or something.

> I'm not clear on how to configure autoupdate with React

Whats confusing about [the Updating docs](https://floating-ui.com/docs/react#updating)?

It takes one line:

```js
useFloating({
  whileElementsMounted: autoUpdate
})
```

> I'm not clear on how to transition tooltips into view - for example in a situation where a single tooltip is shared across elements and it's already open on ElementA, but when ElementB is clicked we want it to fade in, not appear (achievable with keyframes but what a pain, is there a better way?)

Singletons have an open issue: https://github.com/floating-ui/floating-ui/issues/1697. In Tippy they're kinda half-broken and annoying to work with, so a good API needs to be worked out. For now, a `FloatingDelayGroup` replicates their most important functionality (delay grouping)

**Comment by vincerubinetti** [comment]
> We're working on creating something like Tippy for Floating UI though.

Is there a name and repo for that project that I can follow?

What would be great is if it was basically the same API as Tippy but with Floating UI under the hood (for smaller size, modern best practices, etc), or maybe a re-envisioning of it but with the same plug-and-play philosophy.

**Comment by atomiks** [comment]
The `@floating-ui/react` library is that package, there's no vanilla/core version of it atm. 

Tippy's API is not suitable for making actual flexible and accessible floating UI components aside from basic tooltips

**Comment by oje-ext** [comment]
The example in docs are not working. I found this one which does.
